pkg/config/config.go

```diff
@@ -61,6 +61,8 @@ func init() {
 	Datadog.SetEnvPrefix("DD")
 	Datadog.SetTypeByDefaultValue(true)
 
+	Datadog.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
```

Comment:
can you add a small comment explaining what this does? I think it's better to be explicit about the behavior.